lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -161,41 +170,70 @@ struct FirMemory {
            numReadWritePorts == rhs.numReadWritePorts &&
            dataWidth == rhs.dataWidth && depth == rhs.depth &&
            readLatency == rhs.readLatency && writeLatency == rhs.writeLatency &&
-           readUnderWrite == rhs.readUnderWrite;
+           readUnderWrite == rhs.readUnderWrite &&
+           writeUnderWrite == rhs.writeUnderWrite &&
+           writeClockIDs.size() == rhs.writeClockIDs.size() &&
+           llvm::all_of_zip(writeClockIDs, rhs.writeClockIDs,
+                            [](auto a, auto b) { return a == b; });
   }
 };
 } // namespace
 
 static std::string getFirMemoryName(const FirMemory &mem) {
-  return llvm::formatv("FIRRTLMem_{0}_{1}_{2}_{3}_{4}_{5}_{6}_{7}",
-                       mem.numReadPorts, mem.numWritePorts,
-                       mem.numReadWritePorts, mem.dataWidth, mem.depth,
-                       mem.readLatency, mem.writeLatency, mem.readUnderWrite);
+  SmallString<8> clocks;
+  for (auto a : mem.writeClockIDs)
+    clocks.append(Twine((char)(a + 'a')).str());
+  return llvm::formatv(
+      "FIRRTLMem_{0}_{1}_{2}_{3}_{4}_{5}_{6}_{7}_{8}{9}", mem.numReadPorts,
```

Comment:
This is purely for human readability.  If it's getting to ridiculous, it could be shortened.

Reply:
I have some hope that parameterized modules could make this cleaner when they're all ready (not yet)